src/extendedstorage.cpp

```diff
@@ -336,16 +336,18 @@ void ExtendedStorage::unregisterObserver(ExtendedStorageObserver *observer)
 
 void ExtendedStorage::setModified(const QString &info)
 {
-    // Clear all smart loading variables
-    d->mStart = QDate();
-    d->mEnd = QDate();
-    d->mIsUncompletedTodosLoaded = false;
-    d->mIsCompletedTodosDateLoaded = false;
-    d->mIsCompletedTodosCreatedLoaded = false;
-    d->mIsGeoDateLoaded = false;
-    d->mIsGeoCreatedLoaded = false;
-    d->mIsUnreadIncidencesLoaded = false;
-    d->mIsInvitationIncidencesLoaded = false;
+    clearLoaded();
```

Comment:
Noted the clearLoaded() clears a few member variables more. That's probably the correct thing but interesting content on that direction. E.g. mIsDateLoaded which seems to be only used by SqliteStorage::loadIncidences(). 

Description on that method is "Load incidences based on start/due date or creation date." and the code seems to do it only once. But then again a) it seems to be about end date, and b) the query uses the given date so not sure what the member variable protection would be even about.

Not spotting any usage and not sure if this is even usable. Perhaps same applies to some other special cased load methods too.

Potentially a future cleanup candidate?

Reply:
That's correct, some of these flags were not reset in the previous code version.

I think it's an error, and all flags should be reset on storage modification, because all the database content may have been wiped out by the external modification. Who knows ? So all smart loading flags should reset to `false` so any later `load*` call will actually poke the database and not just return.

Reply:
Sure, no disagreement on resetting to false. Mostly just about if the whole member variable, or these load method variants, are sane or usable at all. Now if one calls loadIncidences(true, 1000000, someDate), and then loadIncidences(true, 1000000, someOtherDate), the latter looks like no-op. And that limit handling looks strange too.